macros/edr/data_monitoring/monitors_query/column_monitoring_query.sql

```diff
@@ -1,31 +1,37 @@
 {% macro column_monitoring_query(monitored_table_relation, min_bucket_start, column_obj, column_monitors, metric_properties) %}
-    {% set full_table_name_str = elementary.edr_quote(elementary.relation_to_full_name(monitored_table_relation)) %}
+    {%- set full_table_name_str = elementary.edr_quote(elementary.relation_to_full_name(monitored_table_relation)) %}
+    {%- set min_bucket_start = elementary.edr_date_trunc(metric_properties.time_bucket.period, elementary.edr_cast_as_timestamp(min_bucket_start))%}
```

Comment:
date_trunc by period to the min_bucket start means the it will be rounded to the beginning of the bucket (if week -> first day of week).
It's important for week and month.

Reply:
This means we could be taking a few days back extra in addition to what we previously would have taken. I think it's the better behaviour (better than rounding to the end) but maybe this can mean in some cases we take a lot more days - in case of `month` we can be adding upto 29 more days. 
I think we should add somewhere in the docs of how to use it, also how we round - so users expect this behaviour.

Reply:
If we don't round we will be comparing incomplete buckets to complete buckets (will cause false positives). 
I assume that you would configure month and week buckets expecting us to calculate for full months / weeks. The fact we don't do so is unexpected behavior.

Reply:
I think the end user possibly won't think this all the way through and would maybe be surprised. That's why I suggested adding it in the docs, not doing something else. 
I am thinking about some code running in prod each day with 30 days back, and on the 29th of the month we go back almost twice as much as we do on the 1st of the month. It _is_ a surprise IMO - but whatever you decide about it.

Reply:
It's just on the first execution, minor impact.